de.fu_berlin.inf.dpp/src/de/fu_berlin/inf/dpp/ui/actions/RequestSessionInviteAction.java

```diff
@@ -0,0 +1,81 @@
+package de.fu_berlin.inf.dpp.ui.actions;
+
+import de.fu_berlin.inf.dpp.SarosConstants;
+import de.fu_berlin.inf.dpp.SarosPluginContext;
+import de.fu_berlin.inf.dpp.communication.extensions.JoinSessionRequestExtension;
+import de.fu_berlin.inf.dpp.net.ITransmitter;
+import de.fu_berlin.inf.dpp.net.xmpp.JID;
+import de.fu_berlin.inf.dpp.net.xmpp.discovery.DiscoveryManager;
+import de.fu_berlin.inf.dpp.session.ISarosSession;
+import de.fu_berlin.inf.dpp.session.ISarosSessionManager;
+import de.fu_berlin.inf.dpp.ui.Messages;
+import de.fu_berlin.inf.dpp.ui.util.selection.SelectionUtils;
+import de.fu_berlin.inf.dpp.ui.util.selection.retriever.SelectionRetrieverFactory;
+import java.util.List;
+import org.eclipse.jface.action.Action;
+import org.eclipse.jface.viewers.ISelection;
+import org.eclipse.ui.ISelectionListener;
+import org.eclipse.ui.IWorkbenchPart;
+import org.picocontainer.annotations.Inject;
+
+/** Action for requesting an invitation to a session from a contact. */
+public class RequestSessionInviteAction extends Action implements Disposable {
+
+  @Inject private ISarosSessionManager sessionManager;
+  @Inject private DiscoveryManager discoveryManager;
+  @Inject private ITransmitter transmitter;
+
+  private ISelectionListener selectionListener =
+      new ISelectionListener() {
+        @Override
+        public void selectionChanged(IWorkbenchPart part, ISelection selection) {
+          updateEnablement();
+        }
+      };
+
+  public static final String ACTION_ID = RequestSessionInviteAction.class.getName();
+
+  public RequestSessionInviteAction() {
+    super(Messages.RequestSessionInviteAction_title);
+    setId(ACTION_ID);
+    SarosPluginContext.initComponent(this);
+    SelectionUtils.getSelectionService().addSelectionListener(selectionListener);
+    updateEnablement();
+  }
+
+  @Override
+  public void run() {
+    ISarosSession session = sessionManager.getSession();
+    JID jid = getSelectedJID();
+    if (session != null || jid == null) {
+      return;
+    }
+
+    // TODO: Currently only Saros/S is known to have a working JoinSessionRequestHandler,
```

Comment:
In my opinon, it would be better if this would also be mentioned in the class javadoc.